src/Compilers/CSharp/Portable/Symbols/Metadata/PE/MetadataDecoder.cs

```diff
@@ -230,6 +229,8 @@ public static bool IsOrClosedOverATypeFromAssemblies(TypeSymbol symbol, Immutabl
 
                     do
                     {
+                        Debug.Assert(!namedType.IsTupleType);
```

Comment:
Ok. After more time drilling on this, I was able to hit it after all. 
Added fix and test (`NoPiaEmbedTypes.ErrorType_Tuple`).
Thanks for the prompt.